terraform/environments/digital-prison-reporting/data_ingestion_pipeline.tf

```diff
@@ -0,0 +1,197 @@
+# Lambda which notifies step function when DMS task stops
+module "step_function_notification_lambda" {
+  source = "./modules/lambdas/generic"
+
+  enable_lambda = local.enable_step_function_notification_lambda
+  name          = local.step_function_notification_lambda_name
+  s3_bucket     = local.s3_file_transfer_lambda_code_s3_bucket
+  s3_key        = local.reporting_lambda_code_s3_key
+  handler       = local.step_function_notification_lambda_handler
+  runtime       = local.step_function_notification_lambda_runtime
+  policies      = local.step_function_notification_lambda_policies
+  tracing       = local.step_function_notification_lambda_tracing
+  timeout       = 300 # 5 minutes
+
+  vpc_settings = {
+    subnet_ids = [
+      data.aws_subnet.data_subnets_a.id,
+      data.aws_subnet.data_subnets_b.id,
+      data.aws_subnet.data_subnets_c.id
+    ]
+
+    security_group_ids = [
+      aws_security_group.lambda_generic[0].id
+    ]
+  }
+
+  tags = merge(
+    local.all_tags,
+    {
+      Resource_Group = "ingestion-pipeline"
+      Jira           = "DPR2-209"
+      Resource_Type  = "Lambda"
+      Name           = local.step_function_notification_lambda_name
+    }
+  )
+
+  depends_on = [
+    aws_iam_policy.kms_read_access_policy,
+    aws_iam_policy.dynamodb_access_policy,
+    aws_iam_policy.all_state_machine_policy
+  ]
+}
+
+module "step_function_notification_lambda_trigger" {
```

Comment:
The cloudwatch event to trigger the notification lambda when the DMS task is finished